src/entries/popup/pages/home/Activity/ActivitiesList.tsx

```diff
@@ -48,9 +58,71 @@ export function Activities() {
   });
   useTransactionListForPendingTxs();
   const containerRef = useContainerRef();
+  const { currentAddress } = useCurrentAddressStore();
+  const { currentCurrency } = useCurrentCurrencyStore();
+  const { rainbowChains } = useRainbowChains();
+  const supportedMainnetIds = SUPPORTED_MAINNET_CHAINS.map((c: Chain) => c.id);
+  const { userChains } = useUserChainsStore();
+  const { isWatchingWallet } = useWallets();
+
+  const chainIds = rainbowChains
+    .filter((c) => supportedMainnetIds.includes(c.id) && userChains[c.id])
+    .map((c) => c.id);
+
+  const { data: approvals } = useApprovals({
+    address: currentAddress,
+    chainIds: chainIds,
+    currency: currentCurrency,
+  });
+
+  const tokenApprovals = useMemo(
+    () =>
+      approvals
+        ?.map((approval) =>
+          approval.spenders.map((spender) => ({
+            approval,
+            spender,
+          })),
+        )
+        .flat(),
+    [approvals],
+  );
 
   useActivityShortcuts();
 
+  const isRevokableTransaction = useCallback(
+    (tx: RainbowTransaction) => {
+      if (tx.type === 'approve' && !isWatchingWallet) {
+        const txApproval = tokenApprovals?.find((approval) =>
+          isLowerCaseMatch(approval.spender.tx_hash, tx.hash),
+        );
+        if (txApproval) {
+          return true;
+        }
+      }
+      return false;
```

Comment:
it's just my opinion but I would find something like this easier to read
```suggestion
      if (tx.type !== 'approve' || !isWatchingWallet) return false
      return tokenApprovals?.some((approval) =>
        isLowerCaseMatch(approval.spender.tx_hash, tx.hash),
      );
```